pkg/segment/utils/segconsts.go

```diff
@@ -799,6 +799,55 @@ type CValueEnclosure struct {
 	CVal  interface{}
 }
 
+func (e *CValueEnclosure) Equal(other *CValueEnclosure) bool {
+	if e.Dtype != other.Dtype {
+		return false
+	}
+
+	switch e.Dtype {
+	case SS_DT_STRING:
+		return e.CVal.(string) == other.CVal.(string)
+	case SS_DT_BOOL:
+		return e.CVal.(bool) == other.CVal.(bool)
+	case SS_DT_UNSIGNED_NUM:
+		return e.CVal.(uint64) == other.CVal.(uint64)
+	case SS_DT_SIGNED_NUM:
+		return e.CVal.(int64) == other.CVal.(int64)
+	case SS_DT_FLOAT:
```

Comment:
for float do we need an exact equal or an approximate equal ?

Reply:
Good catch. Probably approximate equal since in general we could do some arithmetic to create/modify the float value